interview-questions.md

```diff
@@ -10,58 +10,87 @@ Then, research the question to expand on your answer. Even if you feel you have
 1. What is `this` in JavaScript?
 
   Your answer:
+  'this' references the object that is executing the current function. If the function is a method in an object 'this' refers to that object itself but if the function is not part of an object, 'this' references the global object which is the window object (in browser).  
 
   Researched answer:
+  'this' keyword refers to the object it belongs to. It has different values depending on where it is used:
+  - In a method, 'this' refers to the owner object.
+  - Alone, 'this' refers to the global object.
+  - In a function, 'this' refers to the global object.
+  - Methods like call(), and apply() can refer 'this' to any object.
 
 
 
 2. What is React? Why would you use it?
 
   Your answer:
+  React is a javascript library that can be reusabled. It used for building/creating user interfaces. 
 
   Researched answer:
+  React is an open-source JavaScript library that is used for building user interfaces specifically for single-page applications. React allows developers to create large web applications that can change data, without reloading the page. The main purpose of React is to be fast, scalable, and simple
 
 
 
 3. Which lifecycle method is required in a React class component?
 
   Your answer:
+  The render() method. It needs to be called in order to return the result/output.
 
   Researched answer:
+  The render() method is the only required method in a class component.
+  When called, it should examine this.props and this.state and return one of the following types: React elements, Arrays and fragments, Portals, String and numbers and Boolean or null.
+
 
 
 
 4. What is JSX? What is one notable difference between HTML and JSX?
 
   Your answer:
+  JSX (JavaScript XML) is a javascript language syntax that allows developers to directly write HTML in React.
+  One notable difference between HTML and JSX is JSX needs to return a parent element using a fragment <> </> wrap around the HTML code block but HTML doesn't need a return.
 
   Researched answer:
+  JSX (JavaScript + XML) is an extension of JavaScript that allows you to write HTML within JavaScript, which has a few benefits of making your code more readable and exercising the full power of JavaScript within HTML.
+  One of the major differences between HTML and JSX is that in JSX, you must return a single parent element, or it won't compile.
+
+
 
 
 
 5. What is yarn? What file(s) are modified in a React application when you run yarn?
 
   Your answer:
+  Yarn is a javascript package manager. It's used to communicate with the repository to help install packages. React uses Node.js to open the app on localhost:3000 and run the start script command on the terminal with yarn. 
 
   Researched answer:
+  Yarn is a software packaging system developed in 2016 by Facebook for Node.js JavaScript runtime environment that provides speed, consistency, stability, and security as an alternative to npm (package manager).
```

Comment:
Yes, yarn will connect dependencies. The yarn.lock, package.json, and node_nodules will be updated.